src/sonic-yang-models/yang-models/sonic-mclag.yang

```diff
@@ -0,0 +1,171 @@
+module sonic-mclag {
+	namespace "http://github.com/Azure/sonic-mclag";
+	prefix smclag;
+	yang-version 1.1;
+
+
+    import ietf-yang-types {
+        prefix yang;
+    }
+
+	import ietf-inet-types {
+		prefix inet;
+	}
+
+    import sonic-types {
+        prefix stypes;
+    }
+
+    import sonic-port {
+        prefix prt;
+    }
+
+    import sonic-portchannel {
+        prefix spc;
+    }
+
+    import sonic-vlan {
+        prefix svlan;
+    }
+
+	organization
+		"SONiC";
+
+	contact
+		"SONiC";
+
+	description
+		"SONIC MCLAG";
+
+	revision 2021-07-28 {
+		description
+			"Addressed review comments";
+	}
+	revision 2019-10-01 {
+		description
+			"Initial revision.";
+	}
+
+	container sonic-mclag {
+
+        container MCLAG_DOMAIN {
+
+            list MCLAG_DOMAIN_LIST { 
+                key "domain_id";
+                max-elements 1;
+
+                leaf domain_id {
+                    description  "MCLAG Domain ID";
+                     type uint16 {
+                        range "1..4095" {
+                            error-message "MCLAG Domain ID out of range";
+                        }
+                    }
+                }
+
+                leaf source_ip {
+                    type inet:ipv4-address;
+                    description  "MCLAG Domain source ip address for session between MCLAG Peers";
+                }
+                leaf peer_ip {
+                    type inet:ipv4-address;
+                    description  "MCLAG Domain peer ip address for session between MCLAG Peers";
+                }
+                leaf peer_link {
+                    description  "MCLAG Domain peer link, data backup path link between MCLAG peers";
+                    type union {
+                        type leafref {
+                            path "/prt:sonic-port/prt:PORT/prt:PORT_LIST/prt:name";
```

Comment:
Add MCLAG config at https://github.com/Azure/sonic-utilities/blob/679a4ba602cc4c1ea77b52e1c782e8663d99e03b/tests/config_dpb_test.py#L715 
to test impact on DPB. Thx